### DISCUSSION TITLE
Discord Webhook, Player Join and Leave message

### DISCUSSION THREAD

**Discussion by TheKogoro** [thijsvanloef/palworld-server-docker#518] [Q&A] [answered]
Hello, 

I have sucessfull make the webhook. i get message when the server stop & start but i was unable to make working player join and leave messages. i am sure at 100% i dont put proprelly the "player_name" function on the stacks file. 

Here how i write that:

   
`       

        - DISCORD_PLAYER_JOIN_MESSAGE=player_name à rejoins le serveur!
        - DISCORD_PLAYER_JOIN_MESSAGE_ENABLED=true
        - DISCORD_PLAYER_JOIN_MESSAGE_URL=https://discord.com/api/webhooks...
        - DISCORD_PLAYER_LEAVE_MESSAGE=player_name à quitté le serveur!
        - DISCORD_PLAYER_LEAVE_MESSAGE_ENABLED=true
        - DISCORD_PLAYER_LEAVE_MESSAGE_URL=https://discord.com/api/webhooks...`

Also wanted to know if it's possible to add a Message POST START of the server (Like : - DISCORD_PRE_START_MESSAGE=server is booting) and after a thing like that: - DISCORD_STARTED_MESSAGE=Server is Online (note it's an exemple) ) because i see when is on pre-start, if i check the logs the server hast started but not finished his full start.
        
Thanks in advance :)

EDIT: Just add that depend how i reboot the server, i dont get the "Shutdown in progress" and "server stopped" message in my discord. for example, if i reboot from Portainer, in general i get all the status reboot messages, if i use the Palbot on discord by using Rcon i get a message ingame but on discord, i get only the Bootup message, no shutdown message (in the logs i can see it have "Sending Discord json: {"embeds":[{"title":"Le serveur est arrêté!","color":14614528}]}" but on my discord i get nothing , i think the message is send too late after the connection between the server and discord is closed)

**Comment by TheKogoro** [comment]
Ok, was able to Make it work. Oddly, somes settings need to redeploy the container and update Stacks itself isnt working for all settings.

**Comment by TheKogoro** [comment]
Hello, I also want to know , after the last update, player join and leave (on logs, on discord and message on the server) i get when player left or join: Player name, (the ID of the save of the user), steam ID. before i get only the User Name when a player letf or join, and Save ID & Steam id is visible for everyone on the server when a player log in/out and on discord. did have anny setting possibility for have only the Username visible ?

**Reply by thijsvanloef** [reply, marked as answer]
@TheKogoro That was a bug, if you upgrade your container to 0.36.0, that should be resolved

**Reply by TheKogoro** [reply]
I have redeploy the container with Portainer and now it's working better, thanks ;)

**Reply by TheKogoro** [reply]
Hello @thijsvanloef , it Is a way to check the container version for See If it match to the last version on GitHub ? (Note i use Portainer for managing container)

Also, just for be sure, for updating the container, i can "redeploy" the container under the Stacks editor. But If i didnt change anything, i can also clic on "recreate" under the container ? (If you have better "how to do" for update the container under Portainer i will Happy to know it :) ).

**Reply by thijsvanloef** [reply]
@TheKogoro I've setup a portainer instance to test this, "recreating" the container using portainer will update the image as long as you've selected "Re-pull image".
![image](https://github.com/thijsvanloef/palworld-server-docker/assets/58031337/fb69d1a1-8266-47a7-929b-f379a58c78f9)
